[docs] Add tablespace and partitioned index content (#11959)
yugabyte/yugabyte-db: docs/content/preview/explore/ysql-language-features/advanced-features/partitions.md

CREATE TABLE order_changes_default PARTITION OF order_changes DEFAULT;
```

> Optionally, you can create indexes on a partitioned table by executing the following command:

Reviewer: I'm sure this was a suggestion from an earlier review, but "as follows" is shorter. You can even omit it, too, if you like.

```suggestion
Optionally, you can create indexes on a partitioned table as follows:

```
